### ISSUE TITLE
Foreign key constraints not enforced in SQL MERGE()

### ISSUE THREAD

**Issue by zachmu** [dolthub/dolt#1773]
Repro:

```sql
id INT(6) UNSIGNED PRIMARY KEY,
  name VARCHAR(30) NOT NULL
);

CREATE TABLE animal (
  id INT(6) UNSIGNED PRIMARY KEY,
  name VARCHAR(30) NOT NULL,
  owner INT(6) UNSIGNED,
  FOREIGN KEY (owner) REFERENCES human(id)
);

SELECT DOLT_COMMIT("-a", "--author", "test <test@example.com>", "-m", "setup database tables");

--------------------------------------------------------------------------------

INSERT INTO human (id, name)
VALUES
  (1, "alice"),
  (2, "bob");

INSERT INTO animal (id, name, owner)
VALUES
  (1, "alice-dog", 1),
  (2, "bob-cat", 2);


SELECT DOLT_COMMIT("-a", "--author", "test <test@example.com>", "-m", "insert test data");

--------------------------------------------------------------------------------

INSERT INTO dolt_branches (name, hash) VALUES ("test_delete", HASHOF("master"));
INSERT INTO dolt_branches (name, hash) VALUES ("test_update", HASHOF("master"));

--------------------------------------------------------------------------------

SELECT DOLT_CHECKOUT('test_delete');

DELETE FROM animal
WHERE owner = 1;

DELETE FROM human
WHERE id = 1;

SELECT DOLT_COMMIT("-a", "--author", "test <test@example.com>", "-m", "delete alice and her pet");

--------------------------------------------------------------------------------

SELECT DOLT_CHECKOUT('test_update');

UPDATE animal
SET name = "alice-puppy"
WHERE owner = 1;

SELECT DOLT_COMMIT("-a", "--author", "test <test@example.com>", "-m", "rename alice's pet");


UPDATE animal
SET id = 1, name = 'alice-puppy', owner = 1
WHERE id = 1;

DELETE FROM dolt_conflicts_animal WHERE base_id = 1;

SET @@dolt_head=COMMIT('--author', "test <test@example.com>", '-m', 'Resolved conflicts');
UPDATE dolt_branches SET hash=@@dolt_head WHERE name="test_conflict";

--------------------------------------------------------------------------------

select * from human;
select * from animal;
select * from animal LEFT JOIN human on animal.owner=human.id;
```

This behavior is unspecified. `dolt merge` on the command line ignores foreign key constraints and expects you to run `dolt verify-constraints` after the fact to enforce them. In SQL, this may not be appropriate.

We need a SQL merge path that allows foreign key enforcement to proceed via the standard mechanisms (system variables), and this behavior needs to be mirrored in some way on the command line.

**Comment by zachmu** [comment]
@andrew-wm-arthur is working on a design for this now, ETA July 1.

**Comment by itsdani** [comment]
Hello!
Do you have any updates on this?

**Comment by andrew-wm-arthur** [comment]
Hey @itsdani! We currently have work in review that addresses this issue: 
https://github.com/dolthub/dolt/pull/1852

**Comment by VinaiRachakonda** [comment]
Merged with #1852
